backend/src/main/java/com/bakdata/conquery/models/forms/managed/EntityDateQueryPlan.java

```diff
@@ -1,77 +1,70 @@
 import lombok.RequiredArgsConstructor;
 
-import javax.validation.Valid;
-import javax.validation.constraints.NotEmpty;
-import javax.validation.constraints.NotNull;
-import java.time.LocalDate;
 import java.util.ArrayList;
-import java.util.Collection;
 import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
 
 /**
  * Implementation of the QueryPlan for an {@link EntityDateQuery}.
  */
 @RequiredArgsConstructor
-public class EntityDateQueryPlan implements QueryPlan {
+public class EntityDateQueryPlan implements QueryPlan<MultilineEntityResult> {
 
 
     private final QueryPlan query;
     private final ArrayConceptQueryPlan features;
     private final List<ExportForm.ResolutionAndAlignment> resolutionsAndAlignments;
-    private final CDateRange dateRange;
+    private final CDateRange dateRestriction;
+
+    private Function<MultilineEntityResult, CDateSet> validityDateCollector;
 
     @Override
-    public EntityResult execute(QueryExecutionContext ctx, Entity entity) {
+    public Optional<MultilineEntityResult> execute(QueryExecutionContext ctx, Entity entity) {
         // Execute the prerequisite query
-        EntityResult preResult = query.execute(ctx, entity);
-        if (preResult.isFailed() || !preResult.isContained()) {
-            return preResult;
+        Optional<EntityResult> preResult = query.execute(ctx, entity);
+        if (preResult.isEmpty()) {
+            return Optional.empty();
         }
         final List<Object[]> resultLines = new ArrayList<>();
-        // It might be a multi line result, so iterate over every result
-        for (Object[] line : preResult.asContained().listResultLines()) {
 
-            // Transform the date set of a result line back to the actual result line
-            CDateSet entityDate = (CDateSet) line[0];
-            entityDate.retainAll(dateRange);
+        CDateSet entityDate = query.getValidityDates(preResult.get());
+        entityDate.retainAll(dateRestriction);
 
-            // Generate DateContexts in the provided resolutions
-            List<DateContext> contexts = new ArrayList<>();
-            for (CDateRange range : entityDate.asRanges()) {
-                contexts.addAll(DateContext.generateAbsoluteContexts(range, resolutionsAndAlignments));
-            }
+        // Generate DateContexts in the provided resolutions
+        List<DateContext> contexts = new ArrayList<>();
+        for (CDateRange range : entityDate.asRanges()) {
+            contexts.addAll(DateContext.generateAbsoluteContexts(range, resolutionsAndAlignments));
+        }
 
-            FormQueryPlan resolutionQuery = new FormQueryPlan(contexts, features);
-            // We assume the date set to be in the first column, this might be wrong
-            EntityResult result = resolutionQuery.execute(ctx, entity);
+        FormQueryPlan resolutionQuery = new FormQueryPlan(contexts, features);
+        validityDateCollector = resolutionQuery::getValidityDates;

```

Comment:
brauchst du den noch?

Reply:
ist ja jetzt alles in einer klasse

Reply:
Jep, aber den plan erzeuge ich erst in der execution, daher lasse ich es so